lib/azure/resource_management/vnet_config.rb

```diff
@@ -0,0 +1,254 @@
+#
+# Author:: Aliasgar Batterywala (aliasgar.batterywala@clogeny.com)
+#
+# Copyright:: Copyright (c) 2016 Opscode, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'ipaddress'
+
+module Azure::ARM
+  module VnetConfig
+
+    ## lists subnets of only a specific virtual network address space ##
+    def subnets_list_specific_address_space(address_prefix, subnets_list)
```

Comment:
@Vasu1105 regarding comments: I have added comments for each method of the `vnet_config` module to explain the purpose in a detailed statement. regarding method name: `subnets_list_specific_address_space` says subnets list of a specific address space, on the other hand `specific_vnet_address_subnet_list` sounds less readable to me than the current name. Anyways will discuss and decide on the names unanimously.